main.tf

```diff
@@ -96,6 +96,17 @@ resource "aws_autoscaling_group" "this" {
   service_linked_role_arn   = var.service_linked_role_arn
   max_instance_lifetime     = var.max_instance_lifetime
 
+  dynamic "instance_refresh" {
+    for_each = var.instance_refresh_enabled ? [1] : []
+    content {
+      strategy = "Rolling"
+      preferences {
```

Comment:
Please add [instance_warmup](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#instance_warmup) argument.

Reply:
Also, block `preferences` should be `dynamic` since it is optional.

Reply:
done! @spatel96 can you set as resolved?